core/src/view/view.cpp

```diff
@@ -467,6 +470,18 @@ glm::vec2 View::lonLatToScreenPosition(double lon, double lat, bool& clipped) co
     return screenPosition;
 }
 
+LngLat View::screenPositionToLngLat(float x, float y, bool& _intersection) {
+
+    if (m_dirtyMatrices) { updateMatrices(); } // Need the view matrices to be up-to-date
+
+    double dx = x, dy = y;
+    double distance = screenToGroundPlaneInternal(dx, dy);
+    _intersection = (distance >= 0);
+    glm::dvec2 meters(dx + m_pos.x, dy + m_pos.y);
+    glm::dvec2 lngLat = m_projection->MetersToLonLat(meters);
```

Comment:
MetersToLngLat for consistency, since you are already on it :D.
Yes, Projection headers will need to change :|

Reply:
Already working on obliterating the MapProjection class in another branch - I'll save it for that ;)

Reply:
Don't think anyone touched projection classes since I added it in 2014, very beginning of tangram-es :D